crates/nomination/src/types.rs

```diff
@@ -0,0 +1,533 @@
+use crate::{ext, Config, Error, Module};
+use codec::{Decode, Encode, HasCompact};
+use frame_support::{dispatch::DispatchResult, ensure, traits::Currency, StorageMap};
+use sp_arithmetic::FixedPointNumber;
+use sp_core::H256;
+use sp_runtime::{
+    traits::{CheckedAdd, CheckedDiv, CheckedSub, Saturating},
+    DispatchError,
+};
+use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
+
+#[cfg(test)]
+use mocktopus::macros::mockable;
+use vault_registry::VaultStatus;
+
+pub(crate) type DOT<T> = <<T as collateral::Config>::DOT as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub(crate) type UnsignedFixedPoint<T> = <T as fee::Config>::UnsignedFixedPoint;
+pub(crate) type Inner<T> = <<T as fee::Config>::UnsignedFixedPoint as FixedPointNumber>::Inner;
+
+pub struct RichOperator<T: Config> {
```

Comment:
I'll leave it like this for now. I can change this in the next PR based on @gregdhill 's input

Reply:
How do you mean to "merge"?

Reply:
I mean that instead of having a struct `RichOperator` with all the methods, and a struct `DefaultOperator` with all the data, why not have a single struct `Operator` with both the data and the methods?